src/Common/src/Interop/Unix/libc/Interop.getcwd.cs

```diff
@@ -10,7 +10,57 @@ internal static partial class Interop
 {
     internal static partial class libc
     {
+        internal static unsafe string getcwd()
+        {
+            const int StackLimit = 256;
+
+            // First try to get the path into a buffer on the stack
+            byte* stackBuf = stackalloc byte[StackLimit];
+            string result = getcwd(stackBuf, StackLimit);
+            if (result != null)
+                return result;
+
+            // If that was too small, try increasing large buffer sizes
+            // until we get one that works or until we hit MaxPath.
+            int maxPath = Interop.libc.MaxPath;
+            int bufferSize = StackLimit;
+            while ((bufferSize = Math.Min(bufferSize * 2, maxPath)) < maxPath)
+            {
+                byte[] buf = new byte[bufferSize];
```

Comment:
I'd considered that, but was scared off by the note that "POSIX.1-2001 leaves the behavior of getcwd() unspecified if buf is NULL."  The Linux, OS X, and FreeBSD man pages all imply that NULL is supported, but still.  I'm going to leave this as-is for now.  If it becomes an issue, we can easily update to it.